Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -162,7 +163,7 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
   const [previousBotPublishHistoryList, setPreviousBotPublishHistoryList] = useState(botPublishHistoryList);
   // check history to see if a 202 is found
   useEffect(() => {
-    // most recent item is a 202, which means we should poll for updates...
+    let publishFinished = true;
```

Comment:
It's better to caculate this boolean value independently via `selectedBots.some(...)` for maintainability, even though it causes an extra iteration.

Reply:
Agree, use built-in primitives whenever you can.

Reply:
move it independently